aggregation-worker/src/main/java/au/org/emii/geoserver/client/HttpIndexReader.java

```diff
@@ -0,0 +1,219 @@
+                    timeField, timeCoverageStart, timeField, timeCoverageEnd
+            );
+
+            Map<String, String> params = new HashMap<String, String>();
+            params.put("typeName", profile);
+            params.put("SERVICE", "WFS");
+            params.put("outputFormat", "csv");
+            params.put("REQUEST", "GetFeature");
+            params.put("VERSION", "1.0.0");
+            params.put("CQL_FILTER", cqlFilter);
+
+            byte[] postDataBytes = encodeMapForPostRequest(params);
+
+            URL url = new URL(downloadUrl);
+            HttpURLConnection conn = null;
+            InputStream inputStream = null;
+            BufferedInputStream bufferedStream = null;
+            DataInputStream dataInputStream = null;
+
+            try
+            {
+                //  Make HTTP request to geoserver
+                conn = (HttpURLConnection) url.openConnection();
+                conn.setRequestMethod("GET");
+                conn.setRequestProperty("Content-Type", "application/x-www-form-urlencoded");
+                conn.setRequestProperty("Content-Length", String.valueOf(postDataBytes.length));
+                conn.setDoOutput(true);
+                conn.getOutputStream().write(postDataBytes);
+                int responseCode = conn.getResponseCode();
+                String responseMessage = conn.getResponseMessage();
+
+                logger.info("HTTP Response : Code [" + responseCode + "], Message [" + responseMessage + "]");
+
+                inputStream = conn.getInputStream();
+                bufferedStream = new BufferedInputStream(inputStream);
+                dataInputStream = new DataInputStream(bufferedStream);
+
+                logger.info(String.format("Getting list of files from '%s'", downloadUrl));
+                logger.info(String.format("Parameters: '%s'", new String(postDataBytes)));
+
+                //  Read response from geoserver - CSV format
+                //  Example structure:
+                //  acorn_hourly_avg_rot_qc_timeseries_url.fid-7a10c7e5_15e34df590d_2000,939620,IMOS/ACORN/gridded_1h-avg-current-map_QC/ROT/2017/03/31/IMOS_ACORN_V_20170331T233000Z_ROT_FV01_1-hour-avg.nc,128950,"ROT, Rottnest Shelf",2017-03-31T23:30:00,POINT (1 2)
+                String line;
+                Integer i = 0;
+                while ((line = dataInputStream.readLine()) != null) {
+                    if (i > 0) { // Skip first line - it's the headers
+                        logger.debug("CSV line    = " + line);
+                        String[] lineParts = line.split(",");
+                        uriList.add(new URI(lineParts[2]));
+                    } else {
+                        logger.debug("CSV headers = " + line);
+                    }
+                    i++;
+                }
+            }
+            finally
+            {
+                if(inputStream != null) { inputStream.close(); }
```

Comment:
Can we use try with resources block instead? - its much cleaner

Reply:
Sure.